awscli/customizations/s3/results.py

```diff
@@ -91,7 +94,13 @@ def _on_success(self, future):
 
     def _on_failure(self, future, e):
         result_kwargs = self._on_done_pop_from_result_kwargs_cache(future)
-        self._result_queue.put(FailureResult(exception=e, **result_kwargs))
+        if isinstance(e, CancelledError):
+            error_result_cls = ErrorResult
+            if 'KeyboardInterrupt' in str(e):
```

Comment:
This seems fragile, shouldn't `s3transfer` expose a better way to disambiguate between these?  What about a `UserCanelledError`?